library/std/src/sys/windows/compat.rs

```diff
@@ -7,47 +7,17 @@
 //! `GetModuleHandle` and `GetProcAddress` to look up DLL entry points at
 //! runtime.
 //!
-//! This implementation uses a static initializer to look up the DLL entry
-//! points. The CRT (C runtime) executes static initializers before `main`
-//! is called (for binaries) and before `DllMain` is called (for DLLs).
-//! This is the ideal time to look up DLL imports, because we are guaranteed
-//! that no other threads will attempt to call these entry points. Thus,
-//! we can look up the imports and store them in `static mut` fields
-//! without any synchronization.
+//! This is implemented simply by storing a function pointer in an atomic
+//! and using relaxed ordering to load it. This means that calling it will be no
```

Comment:
Tragically, we can't use relaxed as mentioned. On x86 the correct orderings for this are free and even on other targets its unlikely to matter compared to the overhead of the syscall.